includes/class-omise-money.php

```diff
@@ -0,0 +1,56 @@
+<?php
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * @since 3.5
+ */
+class Omise_Money {
+	/**
+	 * @var array
+	 */
+	private static $subunit_multiplier = array(
+		'AUD' => 100,
+		'CAD' => 100,
+		'CHF' => 100,
+		'CNY' => 100,
+		'DKK' => 100,
+		'EUR' => 100,
+		'GBP' => 100,
+		'HKD' => 100,
+		'JPY' => 1,
+		'MYR' => 100,
+		'SGD' => 100,
+		'THB' => 100,
+		'USD' => 100
+	);
+
+	/**
+	 * @param  int|float $amount
+	 *
+	 * @return float
+	 */
+	public static function purify_amount( $amount ) {
+		if ( ! is_numeric( $amount ) ) {
+			throw new Exception( __( 'An amount has to be integer, or float.', 'omise' ) );
+		}
+
+		return (float) $amount;
+	}
+
+	/**
+	 * @param  int|float $amount
+	 * @param  string    $currency
+	 *
+	 * @return int
+	 */
+	public function to_subunit( $amount, $currency ) {
+		$amount   = static::purify_amount( $amount );
+		$currency = strtoupper( $currency );
+
+		if ( ! isset( static::$subunit_multiplier[ $currency ] ) ) {
```

Comment:
I'm against having any currency checks here. This is handled elsewhere and is unnecessary here. Adding it here reduces the function's utility

Reply:
Unnecessary is understandable, but why do you think it will reduce the function's utility?
Isn't that utility should not rely on any outside its context (aka. class) to handle something for them? Or else the code will be dependent with that particular code structure?

Also, I think the currency check purpose here is different from ones in each payment method classes.
Those in each payment method classes are to check if the currencies are applicable for a specific method. While here is to check if a given currency can be converted to subunit.

Reply:
It will reduce the function's utility as without the restriction, the function can handle many many more cuurencies. However, a handful of currencies would be wrong. If we switch to this:

```php
class OmiseUtils {
	
	private static
		$currencyDecimals = [
			"BIF" => 0,
			"CLP" => 0,
			"DJF" => 0,
			"GNF" => 0,
			"ISK" => 0,
			"JPY" => 0,
			"KMF" => 0,
			"KRW" => 0,
			"PYG" => 0,
			"RWF" => 0,
			"UGX" => 0,
			"UYI" => 0,
			"VND" => 0,
			"VUV" => 0,
			"XAF" => 0,
			"XOF" => 0,
			"XPF" => 0,
			"BHD" => 3,
			"IQD" => 3,
			"JOD" => 3,
			"KWD" => 3,
			"LYD" => 3,
			"OMR" => 3,
			"TND" => 3,
		],
		$defaultCurrencyDecimals = 2
	;
	
	public static function amountToSubunits($amount, $curr) {
		return $amount * self::subUnitsFor($curr);
	}
	
	public static function subunitsToAmount($subUnits, $curr) {
		return $subUnits / self::subUnitsFor($curr);
	}
	
	public static function subUnitsFor($curr) {
		$c = strtoupper($curr);
		$decimals = isset(self::$currencyDecimals[$c]) ? self::$currencyDecimals[$c] : self::$defaultCurrencyDecimals;
		return pow(10, $decimals);
	}
	
}
```

We cover all currently used currencies worldwide. I think this class should be purely about money/currency and not be bound in any way by any restrictions Omise has. They really don't belong here

Reply:
Isn't it over-design?

Reply:
I think it's actually a simpler design than you propose (i.e. not listing detail for every currency) , and can do more

Reply:
The decimals thing was just to reduce having to type loads of zeros repeatedly